Google.Api.Gax.Grpc/ClientBuilderBase.cs

```diff
@@ -198,6 +210,30 @@ protected void CopySettingsForEmulator(ClientBuilderBase<TClient> source)
             Logger = source.Logger;
         }
 
+        /// <summary>
+        /// Returns the effective settings for this builder, taking into account API keys and any other properties
+        /// which may require additional settings (typically via <see cref="ServiceSettingsBase.CallSettings"/>).
+        /// </summary>
+        /// <remarks>This method only needs to be called if the concrete builder type knows that the settings may
+        /// need to be modified (e.g. if the API supports API keys). It should typically be called as
+        /// <c>GetEffectiveSettings(Settings?.Clone())</c>.</remarks>
+        /// <typeparam name="T">The concrete settings type, derived from <see cref="ServiceSettingsBase"/>, with a
+        /// parameterless constructor that can be used to construct a new default instance.</typeparam>
+        /// <param name="settings">A clone of the existing settings specified in the concrete builder type. May be null.</param>
```

Comment:
I've been thinking about this from the perspective of creating clones unnecessarily... and I'm now not that bothered. It's only once per client creation *and* it's only if the user has specified settings explicitly (which will be pretty rare).